data/yam/agama/auto/template.jsonnet

```diff
@@ -0,0 +1,12 @@
+local lib = import 'lib/base.libsonnet';
+
+function(storage='default', product='SLES', scripts='default', encrypted=false) {
+  product: lib.getProduct(product),
+  user: lib['user'],
+  root: lib['root'],
+  [if scripts != '' then 'scripts']: lib.scripts[scripts],
+
+  [if storage == 'lvm' then 'storage']: lib.storage.lvm(encrypted),
+  [if storage == 'root_filesystem_ext4' then 'storage']: lib.storage['root_filesystem_ext4'],
+  [if storage == 'root_filesystem_xfs' then 'storage']: lib.storage['root_filesystem_xfs'],
+}
```

Comment:
I was expecting to see in this file the possibility to add `pre` and `post` scripts in a flexible way, actually we have right now a case for "import profile" that add only scripts, so in a way `user` and `root` are also optional.